### ISSUE TITLE
Flux output is blurry compared to webui-forge

### ISSUE THREAD

**Issue by Foul-Tarnished** [comfyanonymous/ComfyUI#4514]
I use your workflow (https://comfyanonymous.github.io/ComfyUI_examples/flux/)
but with the GGUF extension (https://github.com/city96/ComfyUI-GGUF)
Q6_K and t5xxl fp8

**I tried to replicate a gen on forge to compare, and it seems it's not just a model/prompt issue:**
(Check in fullscreen to better see)

ComfyUI
<img src="https://github.com/user-attachments/assets/9f2d92d4-978c-44b4-8242-2dc85cb298f6" width="50%"/>

SDWebUI-Forge
<img src="https://github.com/user-attachments/assets/c079f8e5-b323-400e-af22-2fff302870cc" width="50%"/>

It's strange because even though half the generations are blurry on both (model/prompt issue), comfy does appear blurrier, *always**.
It might be even more visible on sharper image (better prompt)

### Other
- Fresh ComfyUI standalone install, then upgraded to Pytorch 12.4, on RTX 4080
- I tried everything I could think of to replicate (CPU noise on forge, same sampling method and schedule type, same sampling steps)
- Could be related to the GGUF extension, as I didn't check on FP8/FP16

**Comment by deepfree2023** [comment]
These were two different pics, not blurry copy of the other.

**Comment by Foul-Tarnished** [comment]
Because one is made in Forge and the other in comfy

I forgot to check the comfy compat setting in forge (but did change to cpu
noise random generator)

doesn't change the fact comfy is always blurrier

Le mer. 21 août 2024, 15:34, deepfree2023 ***@***.***> a
écrit :

**Comment by deepfree2023** [comment]
Have you tried other resolutions? I tested some SDXL recommended resolutions without  blurry results.

**Comment by deepfree2023** [comment]
Tried some higher resolutions, blurry occured. Seems comfyUI support for higher resolution is buggy.

**Comment by ltdrdata** [comment]
When I reproduced using the CPU seed, I obtained completely identical results.

![comfyui-workflow](https://github.com/user-attachments/assets/a13b194a-6855-450c-b6db-9a2e6ecfbf7e)

![forge](https://github.com/user-attachments/assets/472c8ca0-da2e-4a08-bd18-6a37fe9a6ce9)

ComfyUI:
![comfyui](https://github.com/user-attachments/assets/adb2ed42-da4e-4f1c-9b22-b0c43085924b)


Forge:
![forge](https://github.com/user-attachments/assets/61783607-6f7b-41c0-98da-3c433715e3f5)


Both were reproduced in the latest version.

**Comment by deepfree2023** [comment]
Replacing RandomNoise node by RandomNoise (Inspire) node in Foul-Tarnished's comfyUI workflow, and set noise_mode to cpu, the result was just the same as using RandomNoise node. 

So It's already cpu seed in the comfyUI workflow.

**Comment by deepfree2023** [comment]
This workflow can be loaded to replicate the blurry ComfyUI image:
![workflow](https://github.com/user-attachments/assets/b1b93c12-a4d8-4185-adbc-3aa384cfd7d8)

It would be interesting to find out how can forge produce a sharper image without much detail difference to the blurry one?

I've heard some people reporting higher probability of producing blurry images using dev model, and encounted this kind of problem myself.

**Comment by ltdrdata** [comment]
> This workflow can be loaded to replicate the blurry ComfyUI image: ![workflow](https://private-user-images.githubusercontent.com/158825912/360304817-b1b93c12-a4d8-4185-adbc-3aa384cfd7d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMDc1MjQsIm5iZiI6MTcyNDMwNzIyNCwicGF0aCI6Ii8xNTg4MjU5MTIvMzYwMzA0ODE3LWIxYjkzYzEyLWE0ZDgtNDE4NS1hZGJjLTNhYTM4NGNmZDdkOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMlQwNjEzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODkyM2RkOTQxN2E4ZDhmY2M1MjU3ZTg4MDBlNDFjMWZhNzgwMmQ1MzEzNDdlYzgyMmM1ZWZkNjgwNGM1YTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4UQ7P40ZOOQdv5XzLwoa_lqsBDFPCuZpu9Ww55JL4sA)
> 
> It would be interesting to find out how can forge produce a sharper image without much detail difference to the blurry one?
> 
> I've heard some people reporting higher probability of producing blurry images using dev model, and encounted this kind of problem myself.

Although I haven't tested it, disabling the `ModelSamplingFlux` node should result in a configuration identical to Forge.

**Comment by deepfree2023** [comment]
> > This workflow can be loaded to replicate the blurry ComfyUI image: ![workflow](https://private-user-images.githubusercontent.com/158825912/360304817-b1b93c12-a4d8-4185-adbc-3aa384cfd7d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMDc1MjQsIm5iZiI6MTcyNDMwNzIyNCwicGF0aCI6Ii8xNTg4MjU5MTIvMzYwMzA0ODE3LWIxYjkzYzEyLWE0ZDgtNDE4NS1hZGJjLTNhYTM4NGNmZDdkOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMlQwNjEzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODkyM2RkOTQxN2E4ZDhmY2M1MjU3ZTg4MDBlNDFjMWZhNzgwMmQ1MzEzNDdlYzgyMmM1ZWZkNjgwNGM1YTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4UQ7P40ZOOQdv5XzLwoa_lqsBDFPCuZpu9Ww55JL4sA)
> > It would be interesting to find out how can forge produce a sharper image without much detail difference to the blurry one?
> > I've heard some people reporting higher probability of producing blurry images using dev model, and encounted this kind of problem myself.
> 
> Although I haven't tested it, disabling the `ModelSamplingFlux` node should result in a configuration identical to Forge. And is it really true that the `beta` schedule was applied in Forge?

Disabled `ModelSamplingFlux` node , by using normal, simple or sgm_uniform, the result will be sharper, although not the same as Tarnished's Forge image, and still somewhat blurry. 

![ComfyUI_00542_](https://github.com/user-attachments/assets/757beebb-102b-482e-a599-78b863116e5f)

**Comment by ltdrdata** [comment]
> > > This workflow can be loaded to replicate the blurry ComfyUI image: ![workflow](https://private-user-images.githubusercontent.com/158825912/360304817-b1b93c12-a4d8-4185-adbc-3aa384cfd7d8.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMDc1MjQsIm5iZiI6MTcyNDMwNzIyNCwicGF0aCI6Ii8xNTg4MjU5MTIvMzYwMzA0ODE3LWIxYjkzYzEyLWE0ZDgtNDE4NS1hZGJjLTNhYTM4NGNmZDdkOC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMlQwNjEzNDRaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0yODkyM2RkOTQxN2E4ZDhmY2M1MjU3ZTg4MDBlNDFjMWZhNzgwMmQ1MzEzNDdlYzgyMmM1ZWZkNjgwNGM1YTZiJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.4UQ7P40ZOOQdv5XzLwoa_lqsBDFPCuZpu9Ww55JL4sA)
> > > It would be interesting to find out how can forge produce a sharper image without much detail difference to the blurry one?
> > > I've heard some people reporting higher probability of producing blurry images using dev model, and encounted this kind of problem myself.
> > 
> > 
> > Although I haven't tested it, disabling the `ModelSamplingFlux` node should result in a configuration identical to Forge. And is it really true that the `beta` schedule was applied in Forge?
> 
> Disabled `ModelSamplingFlux` node , by using normal, simple or sgm_uniform, the result would be sharper, although not the same as Tarnished's Forge image, and still somewhat blurry.
> 
> ![ComfyUI_00542_](https://private-user-images.githubusercontent.com/158825912/360328902-757beebb-102b-482e-a599-78b863116e5f.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjQzMzk2ODEsIm5iZiI6MTcyNDMzOTM4MSwicGF0aCI6Ii8xNTg4MjU5MTIvMzYwMzI4OTAyLTc1N2JlZWJiLTEwMmItNDgyZS1hNTk5LTc4Yjg2MzExNmU1Zi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQwODIyJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MDgyMlQxNTA5NDFaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT01YTQzZTRmYzBkMzBiYjI3YzBmMmVmZDBmNjBhMjNmMGViYTUxNmIyZGRkOTM0M2Q4YzM4NzYzYzZiMGFiYmY0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZhY3Rvcl9pZD0wJmtleV9pZD0wJnJlcG9faWQ9MCJ9.LWMKQ6f97bhUHXBXTKpVzhJ2xY47CnaD5MdNXvZqORM)

In the beta schedule, both resulted in blurred images, while in the simple schedule, completely identical images were obtained.

When applying the beta schedule, I'm not sure what additional steps Forge takes internally, but there were subtle differences in detail. While ComfyUI produced a slightly more blurred image, Forge's output was also blurred.

It's not that the beta schedule has a tendency to create blurred images.
According to my tests, FLUX simply has a tendency to generate blurred images with certain prompts.
And when the results are not exactly the same, it just leads to differently blurred images.

**Comment by deepfree2023** [comment]
Made a few more tests：
1. Dev models, at 976 x 1328, the blurry image ratio is higher(about 3/8-4/8 in my tests), using different scheduler may increase or decrease the blurry level, but can not make the image really sharp, ModelSamplingFlux does not affect the blurry level obviously. Adjust pompt seems cannot 
2. Dev models, at 768 x 1280, the blurry image ratio is lower (about 1/8 in my tests).
3. Schnell, at 976 x 1328 or 768 x 1280,  the blurry image ratio is 0.

**Comment by deepfree2023** [comment]
"FLUX simply has a tendency to generate blurred images with certain prompts." This could be the case. 
It might be that to get better image quality, there's a tendency to use a shallow depth of field, which makes it easier to go out of focus?

**Comment by deepfree2023** [comment]
Tested some fine-tuned model, declaring solved blurry problem, and yes, very sharp images without blurry up to now. But they are still in the early stage, and have other problems. 

This kind of fine-tuning models or loras could be the solution to the blurry image problem.

**Comment by Foul-Tarnished** [comment]
I made some new tests on a fresh install of ComfyUI v0.1.2 (plus ran update.py)
I also updated Forge and set compatibility mode to ComfyUI.

ComfyUI
![ComfyUI_00001_](https://github.com/user-attachments/assets/c8b1a92d-7f52-44a6-959e-0489d8632895)

Forge
![00005-1120098321](https://github.com/user-attachments/assets/93675f7d-5d64-49b1-831b-377d22765076)

Seems to be fixed, idk, I re-used the same workflow from [here](https://comfyanonymous.github.io/ComfyUI_examples/flux/)
Euler, Beta, 40 steps instead of 20

**Comment by frankchieng** [comment]
> I use your workflow (https://comfyanonymous.github.io/ComfyUI_examples/flux/) but with the GGUF extension (https://github.com/city96/ComfyUI-GGUF) Q6_K and t5xxl fp8
> 
> **I tried to replicate a gen on forge to compare, and it seems it's not just a model/prompt issue:** (Check in fullscreen to better see)
> 
> ComfyUI <img alt="" width="50%" src="https://private-user-images.githubusercontent.com/107711110/359693486-9f2d92d4-978c-44b4-8242-2dc85cb298f6.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NjYyMzgsIm5iZiI6MTcyODQ2NTkzOCwicGF0aCI6Ii8xMDc3MTExMTAvMzU5NjkzNDg2LTlmMmQ5MmQ0LTk3OGMtNDRiNC04MjQyLTJkYzg1Y2IyOThmNi5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwOTI1MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT05ZDAxMDRjZjZhZGFlMjdhMWFhZmI2NWNhZjliZTU0OTRmNzJjN2QxYmRmOGQxMjFjM2Q2MGU4N2MxMjk1ZjRjJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9._CLP3Hz_y-GXttPQoiNRYatnX4J_YZ4vv3Nf2_94Syw">
> 
> SDWebUI-Forge <img alt="" width="50%" src="https://private-user-images.githubusercontent.com/107711110/359693545-c079f8e5-b323-400e-af22-2fff302870cc.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjg0NjYyMzgsIm5iZiI6MTcyODQ2NTkzOCwicGF0aCI6Ii8xMDc3MTExMTAvMzU5NjkzNTQ1LWMwNzlmOGU1LWIzMjMtNDAwZS1hZjIyLTJmZmYzMDI4NzBjYy5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMDA5JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTAwOVQwOTI1MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jYWEwNmE1NjZlYTQ5YzM4MjEzNGJmNjkxNDA0YzViOGM3YTgxNTliZjFlN2U0Y2ZlMjVkZmYxM2U1NTA5MjY2JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.lUDk0TGZMBxCZ5ByCsXennMcSaF_Aqjnw8QC8c49ais">
> 
> It's strange because even though half the generations are blurry on both (model/prompt issue), comfy does appear blurrier, _always_*. It might be even more visible on sharper image (better prompt)
> 
> ### Other
> * Fresh ComfyUI standalone install, then upgraded to Pytorch 12.4, on RTX 4080
> * I tried everything I could think of to replicate (CPU noise on forge, same sampling method and schedule type, same sampling steps)
> * Could be related to the GGUF extension, as I didn't check on FP8/FP16

choose beta of ksampler schduler instead of simple

**Comment by shaun-ba** [comment]
I still get this, what's the fix?

**Comment by deepfree2023** [comment]
> I still get this, what's the fix?

Have you tried antiblur lora or something like that?
